### PR TITLE
Mostafa dev

### PR DESCRIPTION
**Pull request by MostafaRahgouy**

### CODE REVIEW

File: word_conditional_model.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        word_matrix = np.zeros((number_of_sentences, number_of_words))
+        return word_matrix
+
+
+    def __represent_sentences_as_matrix(self, matrix, words_lsts):
+        for i in range(len(words_lsts)):
+            for j in range(len(words_lsts[i])):
+                mj = self.__words.index( stemmer().stem(words_lsts[i][j].lower()))
+                for m in range(9):
+                    matrix[(i * 9) + m][mj] = 1
```
Comment (Rahgooy) [inline word_conditional_model.py]: Without the most inner for
```python
   matrix[i*9:(i+1)*9][mj] = 1
```

File: word_conditional_model.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return matrix
+
+    def __represent_bios_as_matrix(self, matrix, bios_lsts, words_lsts):
+        for i in range(len(words_lsts)):
+            for j in range(len(words_lsts[i])):
+                d = self.__words.index( stemmer().stem(words_lsts[i][j].lower()))
+                for m in range(9):
+                    if bios_lsts[i][j][m * 2] == "I" or bios_lsts[i][j][m * 2] == "B":
+                        matrix[(i * 9) + m][d] = 1
```
Comment (Rahgooy) [inline word_conditional_model.py:37]: The same trick as the previous one

File: word_conditional_model.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        for word in words_vocab:
+            if word not in self.__words:
+                self.__words.append(word)
+
+        self.__words_matrix = self.__build_matrix(len(words_lsts * 9), len(self.__words))
+        self.__bios_matrix = self.__build_matrix(len(words_lsts * 9), len(self.__words))
+        self.__model = self.__build_matrix(len(self.__words), len(self.__words))
+
+        self.__words_matrix = self.__represent_sentences_as_matrix(self.__words_matrix, words_lsts)
+        self.__bios_matrix = self.__represent_bios_as_matrix(self.__bios_matrix, bios_lsts, words_lsts)
```
Comment (Rahgooy) [inline word_conditional_model.py]: I think it's better to do not send `self.__words_matrix` as a parameter for these functions. I think moving the `build_matrix` call to the `__represent` function would be nicer.

Comment (MostafaRahgouy) [reply word_conditional_model.py]: Good point! We will do it.

File: word_conditional_model.py (hunk 1 above)
Comment (MostafaRahgouy) [reply word_conditional_model.py]: We did it. Thanks for sharing!

File: word_conditional_model.py (hunk 3 above)
Comment (MostafaRahgouy) [reply word_conditional_model.py]: The code has been changed!

File: word_conditional_model.py (hunk 2 above)
Comment (MostafaRahgouy) [reply word_conditional_model.py:37]: We couldn't use it.

**Comment by Rahgooy** [comment]
Looks good!
Merged.

**Pull request merged**